### PR TITLE
T05

### PR DESCRIPTION
**Pull request by dineshN52**

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -1,7 +1,27 @@
...
-    }
+﻿using System;
+namespace Training {
+   namespace Prime {
+      class Program {
+         static bool IsPrime (int num) {
+            if (num <= 1)
+               return false;
+            else if (num == 2 & num == 3)
+               return true;
```
Comment (ravin-imm) [inline Program.cs:9]: This is always false as the "num" can't be equal to 2 and 3 at the same time. Is this check required?

File: Program.cs [hunk 2]
```diff
@@ -1,7 +1,27 @@
...
+            for (int i = num - 1; i > 1; i--) {
+               if (num % i == 0)
+                  return false;
```
Comment (ravin-imm) [inline Program.cs:12]: Instead of looping through till the number before, can you simplify this?

File: Program.cs [hunk 3]
```diff
@@ -1,7 +1,27 @@
...
+            return true;
+         }
+         static void Main (string[] args) {
+            Console.Write ("Enter a number: ");
+            int number = int.Parse (Console.ReadLine ());
+
+            if (IsPrime (number))
+               Console.WriteLine ($"{number} is a prime number.");
+            else
+               Console.WriteLine ($"{number} is not a prime number.");
```
Comment (ravin-imm) [inline Program.cs:23]: Check if you can use a single _Console.WriteLine (..)_ statement by changing the message depending on the check inside the _Console.WriteLine (..)_ statement itself.

**Review by ravin-imm — CHANGES_REQUESTED** [review]

File: Program.cs (hunk 3 above)
Comment (dineshN52) [reply Program.cs]: Both condition is printed using single Console.Writeline function using ternary operator and code was updated sir

File: Program.cs (hunk 1 above)
Comment (dineshN52) [reply Program.cs]: As the comparison is not required it is removed and updated sir

File: Program.cs [hunk 4]
```diff
@@ -1,7 +1,19 @@
...
-        }
-    }
+﻿using System;
+namespace Prime {
+   class Program {
+      static bool IsPrime (int b) {
+         if (b <= 1)
+            return false;
+         for (int i = b - 1; i > 1;i--)
+            if (b % i == 0)
```
Comment (ravin-imm) [inline Program.cs:8]: Can you simplify the loop count?

**Review by ravin-imm — CHANGES_REQUESTED** [review]

File: Program.cs (hunk 4 above)
Comment (dineshN52) [reply Program.cs]: Loop count reduced by creating loop condition with square of i ,by modular arithmetic b modulo i*i & b modulo i results same.

File: Program.cs [hunk 5]
```diff
@@ -1,7 +1,19 @@
...
+            return false;
+         for (int i = 2; i * i <= b; i++) 
+            if (b % i == 0) 
+               return false;
+         return true;
+      }
+      static void Main (string[] args) {
+         Console.Write ("Enter a number:");
+         int a = int.Parse (Console.ReadLine ());
+         Console.WriteLine (string.Concat ($"{a} is" , IsPrime (a) ? " a prime number" : " not a prime number"));
```
Comment (ravin-imm) [inline Program.cs:15]: Try this out without string.Concat (..)

**Review by ravin-imm — CHANGES_REQUESTED** [review]

File: Program.cs (hunk 5 above)
Comment (dineshN52) [reply Program.cs]: Printed the same with interpolation sir

**Review by ravin-imm — APPROVED** [review]

**Pull request closed without merging**